### PR TITLE
docs: fix lua type warnings

### PR DESCRIPTION
**Pull request by MariaSolOs**
Minor type improvements.

### CODE REVIEW

File: runtime/lua/vim/_meta/builtin_types.lua [hunk 1]
```diff
@@ -127,3 +127,11 @@
...
 --- @field topfill integer
 --- @field topline integer
+
+--- @class vim.fn.getscriptinfo.ret
+--- @field autoload false
+--- @field functions? string[]
+--- @field name string
+--- @field sid string
+--- @field variables? table<string, any>
+--- @field vim 1
```
Comment (zeertzjq) [inline runtime/lua/vim/_meta/builtin_types.lua:137]:
```suggestion
--- @field version 1
```

File: src/nvim/eval.lua [hunk 2]
```diff
@@ -4475,6 +4475,7 @@ M.funcs = {
     ]=],
     name = 'getscriptinfo',
     params = { { 'opts', 'table' } },
+    returns = 'vim.fn.getscriptinfo.ret|vim.fn.getscriptinfo.ret[]',
```
Comment (zeertzjq) [inline src/nvim/eval.lua:4478]: This function also returns a list

Comment (zeertzjq) [inline src/nvim/eval.lua:4478]: This function always returns a list

File: runtime/lua/vim/_meta/builtin_types.lua (hunk 1 above)
Comment (wookayin) [inline runtime/lua/vim/_meta/builtin_types.lua:137]:
```suggestion
--- @field version 1
```

File: src/nvim/eval.lua (hunk 2 above)
Comment (MariaSolOs) [reply src/nvim/eval.lua:4478]: Nope, the docs say that is `sid` is specified, only information about the script with ID `"sid"` is returned. There is even this example: `echo getscriptinfo({'sid': 15}).variables`

Comment (zeertzjq) [reply src/nvim/eval.lua:4478]: Hmm, the docs is wrong. When `sid` specified, it returns a list with a single item.

Comment (MariaSolOs) [reply src/nvim/eval.lua:4478]: Oh yes you're right, I just checked the tests and it always returns a list. Fixing it.

File: src/nvim/eval.lua [hunk 3]
```diff
@@ -4470,11 +4470,12 @@ M.funcs = {
 
       Examples: >vim
       	echo getscriptinfo({'name': 'myscript'})
-      	echo getscriptinfo({'sid': 15}).variables
+      	echo getscriptinfo({'sid': 15})[1].variables
```
Comment (zeertzjq) [inline src/nvim/eval.lua:4473]:
```suggestion
      	echo getscriptinfo({'sid': 15})[0].variables
```

Comment (MariaSolOs) [reply src/nvim/eval.lua:4473]: uh duh Lua's 1-based indexing is getting into me.

**Pull request merged**